lib/resources/gem.rb

```diff
@@ -48,4 +52,48 @@ def to_s
       "gem package #{@package_name}"
     end
   end
+
+  class ChefGemPackage < GemPackage
+    name 'chef_gem'
+    desc 'Use the chef_gem InSpec audit resource to test if a gem is installed in the chef omnibus.'
+    example "
+      describe chef_gem('chef-sugar') do
+        it { should be_installed }
+        its('version') { should eq '3.4.0' }
+      end
+    "
+
+    def initialize(package_name)
+      gem_binary = if inspec.os.windows?
+                     'c:\opscode\chef\embedded\bin\gem'
+                   else
+                     '/opt/chef/embedded/bin/gem'
+                   end
+      super(package_name, gem_binary)
+    end
+
+    def to_s
+      "chef_gem #{@package_name}"
+    end
+  end
+
+  class ChefServerGemPackage < GemPackage
+    name 'chef_server_gem'
```

Comment:
I totally support the changes to the `gem` resource to support alternate gem binaries, and totally support the `chef_gem` resource as an easy shortcut.

However, I wonder if this is a slippery slope and `chef_server_gem` is the top of that slope. 🙂  Given how often InSpec is run via or alongside Chef, `chef_gem` has a lot of use cases. I just wonder if `chef_server_gem` will be used too infrequently and therefore should just use the new extension of `gem` directly.